### PR TITLE
upd934g: use device_rom_interface instead of a read callback

### PR DESCRIPTION
**Pull request by devinacker**

### CODE REVIEW

File: src/devices/sound/upd934g.cpp [hunk 1]
```diff
@@ -117,7 +114,6 @@ void upd934g_device::sound_stream_update(sound_stream &stream, std::vector<read_
 	}
 }
 
-
```
Comment (startaq) [inline src/devices/sound/upd934g.cpp:120]: Please restore this blank line.

File: src/mame/casio/rz1.cpp [hunk 2]
```diff
@@ -136,6 +134,17 @@ void rz1_state::map(address_map &map)
...
+{
+	map(0x0000, 0x7fff).rom();
+	map(0x8000, 0x9fff).ram().share("sample1");
+	map(0xa000, 0xbfff).ram().share("sample2");
+}
+
+void rz1_state::pg1_map(address_map& map)
+{
+	map(0x0000, 0x7fff).rom();
+}
```
Comment (startaq) [inline src/mame/casio/rz1.cpp:148]: Please add a blank line here (two blank lines at the start of a new major section).

**Review by startaq — COMMENTED** [review]
Looks good, just some very minor adjustments. I assume you tested that recording and playing back samples still works (don't see a reason why it shouldn't, but just in case).

File: src/mame/casio/rz1.cpp [hunk 3]
```diff
@@ -95,16 +93,16 @@ class rz1_state : public driver_device
 	output_finder<> m_led_startstop;
 
 	void map(address_map &map);
+	void pg0_map(address_map& map);
+	void pg1_map(address_map& map);
```
Comment (cuavas) [inline src/mame/casio/rz1.cpp:97]: Please follow the style of the rest of the file: right-associative qualifiers on the right of the space, i.e.
```
	void pg0_map(address_map &map);
	void pg1_map(address_map &map);
```

File: src/mame/casio/rz1.cpp [hunk 4]
```diff
@@ -136,6 +134,18 @@ void rz1_state::map(address_map &map)
...
 }
 
+void rz1_state::pg0_map(address_map& map)
+{
+	map(0x0000, 0x7fff).rom();
+	map(0x8000, 0x9fff).ram().share("sample1");
+	map(0xa000, 0xbfff).ram().share("sample2");
+}
+
+void rz1_state::pg1_map(address_map& map)
```
Comment (cuavas) [inline src/mame/casio/rz1.cpp:144]: Same here.

**Pull request merged**